[16.0][MIG] l10n_es_aeat_sii_invoice_summary: Migration to 16.0

RodrigoBM opened the pull request:
#2526

VRodrigo committed:
[ADD] l10n_es_aeat_sii_invoice_summary : añadir resumen de facturas

omar7r committed:
[MIG] l10n_es_aeat_sii_invoice_summary: Backport from 11.0 to 10.0

[ADD] icon.png

[UPD] Update l10n_es_aeat_sii_invoice_summary.pot

[UPD] README.rst

[UPD] README.rst

ramiadavid committed:
[MIG] l10n_es_aeat_sii_invoice_summary: Migration to 14.0

ramiadavid committed:
[IMP] l10n_es_aeat_sii_invoice_summary: añadir rectificativas y tests

SoniaViciana commented:
LGTM!

Envío al SII de factura agrupada de simplificadas:
![image](https://github.com/OCA/l10n-spain/assets/46522913/b151655a-8534-4f4a-b0a8-701dcd606864)


Envío al SII de rectificativa agrupada de simplificadas (la única diferencia es que los importes aparecen en negativo):

![image](https://github.com/OCA/l10n-spain/assets/46522913/458cd2fc-bd13-44df-8f3c-cac1fb86abd8)

Gracias,

ljsalvatierra-factorlibre commented on l10n_es_aeat_sii_invoice_summary/models/account_move.py:
```diff
@@ -13,12 +13,12 @@ class AccountMove(models.Model):
             if self.sii_invoice_summary_start:
                 if self.sii_invoice_summary_start == self.sii_invoice_summary_end:
-                    tipo_factura = "F2" if self.move_type == "out_invoice" else "R5"
+                    tipo_factura = "F2" if self.is_inbound() else "R5"
```
El método incluye más tipos, ¿es correcto?

```python
    @api.model
    def get_inbound_types(self, include_receipts=True):
        return ['out_invoice', 'in_refund'] + (include_receipts and ['out_receipt'] or [])

    def is_inbound(self, include_receipts=True):
        return self.move_type in self.get_inbound_types(include_receipts)
```

ljsalvatierra-factorlibre commented on l10n_es_aeat_sii_invoice_summary/views/account_move_view.xml:
```diff
@@ -9,14 +9,17 @@
                 <field
                     name="sii_invoice_summary_start"
-                    attrs="{'invisible': [('is_invoice_summary', '=', False)]}"
+                    attrs="{'invisible': ['|', ('move_type', 'not in', ('out_invoice','out_refund')), ('is_invoice_summary', '=', False)]}"
```
¿Es necesario esto? Parece redundante.

ljsalvatierra-factorlibre commented:
Hola @OCA/local-spain-maintainers, ¿se podría reabrir el PR por favor?

pedrobaeza commented:
Mencionad por favor en el README que esto es una alternativa al `l10n_es_pos_sii`.

pedrobaeza commented:
/ocabot migration l10n_es_aeat_sii_invoice_summary

victoralmau committed:
[IMP] l10n_es_aeat_sii_invoice_summary: black, isort, prettier

victoralmau committed:
[MIG] l10n_es_aeat_sii_invoice_summary: Finish migration 14.0

TT49186

Commit:
[UPD] Update l10n_es_aeat_sii_invoice_summary.pot

OCA-git-bot committed:
[BOT] post-merge updates

victoralmau committed:
[IMP] l10n_es_aeat_sii_invoice_summary: Show summary fields only on customer invoices

TT49186

OCA-git-bot committed:
[BOT] post-merge updates

victoralmau committed:
[IMP] l10n_es_aeat_sii_invoice_summary: Define a specific string for the fields in form view

TT49186

Commit:
[UPD] Update l10n_es_aeat_sii_invoice_summary.pot

OCA-git-bot committed:
[BOT] post-merge updates

victoralmau committed:
[IMP] l10n_es_aeat_sii_invoice_summary: Changes in the visibility and required of the fields

TT49397

Commit:
[UPD] Update l10n_es_aeat_sii_invoice_summary.pot

OCA-git-bot committed:
[BOT] post-merge updates

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: l10n-spain-14.0/l10n-spain-14.0-l10n_es_aeat_sii_invoice_summary
Translate-URL: https://translation.odoo-community.org/projects/l10n-spain-14-0/l10n-spain-14-0-l10n_es_aeat_sii_invoice_summary/

aritzolea committed:
[IMP] l10n_es_aeat_sii_invoice_summary: pre-commit stuff

aritzolea committed:
[MIG] l10n_es_aeat_sii_invoice_summary: Migration to 16.0

aritzolea commented:
@pedrobaeza Readme actualizado con el comentario de que es una alternativa a `l10n_es_pos_sii` y actualizado el historial de commits del módulo, ahora queda mucho mas simple la migración

pedrobaeza reviewed (approved):
/ocabot merge nobump
